lib/src/diff_presentation/mod.rs

```diff
@@ -0,0 +1,91 @@
+//! Utilities to present file diffs to the user
+#![expect(missing_docs)]
+
+use bstr::BString;
+use pollster::FutureExt as _;
+
+use crate::backend::BackendResult;
+use crate::conflicts::MaterializedFileValue;
+use crate::diff::ContentDiff;
+use crate::diff::find_line_ranges;
+use crate::merge::Merge;
+use crate::repo_path::RepoPath;
+
+pub mod unified;
+// TODO: colored_diffs utils should also be moved from `jj_cli::diff_utils` to
+// here.
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
```

Comment:
nit: at your option, these common types and functions can be extracted to util or something. If all diff presentation functions are moved to jj-lib, this utility module will become private.

Reply:
I think this is probably the right thing to do, but it might make more sense to reorganize it after the other diff presentation functions are moved here. At the moment, I don't have a great sense of what's a utility function, as you can tell by my putting some things into wrong places.

So, I'll punt this off to later.